src/main/java/com/theironyard/MicroblogDbController.java

```diff
@@ -0,0 +1,61 @@
+package com.theironyard;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpSession;
+
+import java.util.List;
+
+
+@Controller
+public class MicroblogDbController {
+    private static String loggedInUser = "loggedInUser";
+    @Autowired
+    MessageRepository messages;
+
+    @RequestMapping(path = "/update", method = RequestMethod.POST)
+    public String update(int id, String text) {
+        Message msg = new Message(id, text);
+        messages.save(msg);//ToDo check this...
```

Comment:
The correct way to update the existing record is to select it from the database. However, that would take an extra step. Your way here is more efficient although it allows the update to any id which could lead to errors. The best practice approach is to create an UPDATE sql statement using the @Query annotation.